src/main/java/bio/terra/service/iam/sam/SamIam.java

```diff
@@ -53,14 +53,18 @@
 
     private final SamConfiguration samConfig;
     private final ConfigurationService configurationService;
+    private final SamRetry samRetry;
 
     @Autowired
     public SamIam(SamConfiguration samConfig,
-                  ConfigurationService configurationService) {
+                  ConfigurationService configurationService,
+                  SamRetry samRetry) {
         this.samConfig = samConfig;
         this.configurationService = configurationService;
+        this.samRetry = samRetry;
```

Comment:
Is this safe to reuse?  I though these retry objects were stateful, in which case we could get in trouble if we run these concurrently, no?

Reply:
Thank you for catching this!
If nothing else, I think the "operationTimeout" gets set when SamRetry is defined. I think we would want this to happen as close to the beginning of the retry as possible.  https://github.com/DataBiosphere/jade-data-repo/blob/develop/src/main/java/bio/terra/service/iam/sam/SamRetry.java#L33